### DISCUSSION TITLE
[@types/react-modal] Unable to use with Typescript

### DISCUSSION THREAD

**Discussion by alexlibbysw** [DefinitelyTyped/DefinitelyTyped#60781] [Issues with a @types package]
I'm trying to use @react-modal in a React-based component library using Typescript - this is generating a host of errors, similar to this:
```
ERROR: src/components/OverlaySheet/OverlaySheet.tsx:10:11 - error TS2430: Interface 'OverlaySheetProps' incorrectly extends interface 'Props'.
Types of property 'children' are incompatible.
Type 'ReactNode | ((api: { onDismiss: () => void | undefined; }) => ReactNode)' is not assignable to type 'ReactNode'.
Type '(api: { onDismiss: () => void | undefined; }) => ReactNode' is not assignable to type 'ReactNode'.
```
I've added a workaround using `const Modal: any = ReactModal;`, but this isn't ideal - I have to apply 2 Typescript directives to stop ESLint from going nuts! I'm pretty sure the cause of my issue is that the `@types/react-modal` file doesn't yet support React 18 - anyone have any update on when it might be updated? The version I use at present is 3.13.1 - I've seen a ticket asking for something similar, but with no response as to when a newer version may be released...

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
I caon only say, the `types/react-modal` were modified 8 months ago ahead for a compatibility with react v18 (`children?: React.ReactNode;` property as added to props). What's the relation of @types/react-modal to an error you've posted?
You can pin down the @types/ract-modal to version 17 in your deps resolution. React modal itself is still on v17

**Comment by NathanVss** [comment]
Some problem here :(

**Reply by NathanVss** [reply]
I found the solution, please see https://github.com/reactjs/react-modal/issues/960#issuecomment-1290775810
